RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/TransformerToIJsonTransformerAdapter.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+
+    @SuppressWarnings("unchecked")
+    private static Map<String, Object> objectNodeToMap(Object node) {
+        return (Map<String, Object>) MAPPER.convertValue(node, Map.class);
+    }
+
+    @SneakyThrows
+    private static String printMap(Map<String, Object> map) {
+        return MAPPER.writeValueAsString(map);
+    }
+
+    @SuppressWarnings("unchecked")
+    private MigrationItem transformMigrationItem(MigrationItem migrationItem) {
+        // Keep untouched original for logging
+        final Map<String, Object> originalMap = MAPPER.convertValue(migrationItem, Map.class);
+        var transformedMigrationItem = transformer.transformJson(MAPPER.convertValue(migrationItem, Map.class));
+        logTransformation(originalMap, transformedMigrationItem);
+        return MAPPER.convertValue(transformedMigrationItem, MigrationItem.class);
+    }
+
+    void updateTemplates(Collection<? extends MigrationItem> transformedItems, ObjectNode itemsRoot) {
+        itemsRoot.removeAll();
+        transformedItems.forEach(item ->
+                {
+                    log.atInfo().setMessage("Setting new item of type {}, name {}, body {}")
+                            .addArgument(item.type)
+                            .addArgument(item.name)
+                            .addArgument(item.body)
+                            .log();
+                    itemsRoot.set(item.name, item.body);
+                }
+        );
+    }
+
+    @Override
+    public GlobalMetadata transformGlobalMetadata(GlobalMetadata globalData) {
+        var inputJson = objectNodeToMap(globalData.toObjectNode());
+        log.atInfo().setMessage("BeforeJsonGlobal: {}").addArgument(() -> printMap(inputJson)).log();
+        var afterJson = transformer.transformJson(inputJson);
+        log.atInfo().setMessage("AfterJsonGlobal: {}").addArgument(() -> printMap(afterJson)).log();
+
+
+        final List<LegacyTemplate> legacyTemplates = new ArrayList<>();
+        globalData.getTemplates().fields().forEachRemaining(
+                entry -> legacyTemplates.add(new LegacyTemplate(entry.getKey(), (ObjectNode) entry.getValue()))
+        );
+        final List<IndexTemplate> indexTemplates = new ArrayList<>();
+        globalData.getIndexTemplates().fields().forEachRemaining(
+                entry -> indexTemplates.add(new IndexTemplate(entry.getKey(), (ObjectNode) entry.getValue()))
+        );
+        final List<ComponentTemplate> componentTemplates = new ArrayList<>();
+        globalData.getComponentTemplates().fields().forEachRemaining(
+                entry -> componentTemplates.add(new ComponentTemplate(entry.getKey(), (ObjectNode) entry.getValue()))
+        );
+
+        var transformedTemplates = Stream.concat(Stream.concat(
+                legacyTemplates.stream(),
+                indexTemplates.stream()),
+                componentTemplates.stream()
+                )
+                .map(this::transformMigrationItem).collect(Collectors.toList());
```

Comment:
Why combine all the different migration item types into an undifferentiated stream, transform, then split them up again?  Wouldn't it be easier and more efficient to just apply the transform to each class individually?